pkg/reconciler/kubernetes/tektondashboard/tektondashboard.go

```diff
@@ -145,11 +313,52 @@ func (r *Reconciler) transform(ctx context.Context, manifest *mf.Manifest, comp
 	return common.Transform(ctx, manifest, instance, extra...)
 }
 
-func (r *Reconciler) installed(ctx context.Context, instance v1alpha1.TektonComponent) (*mf.Manifest, error) {
-	// Create new, empty manifest with valid client and logger
-	installed := r.manifest.Append()
-	// TODO: add ingress, etc
-	stages := common.Stages{common.AppendInstalled, r.transform}
-	err := stages.Execute(ctx, &installed, instance)
-	return &installed, err
+func (r *Reconciler) createInstallerSet(ctx context.Context, td *v1alpha1.TektonDashboard) (*v1alpha1.TektonInstallerSet, error) {
+
+	manifest := r.manifest
+	if err := r.transform(ctx, &manifest, td); err != nil {
+		td.Status.MarkNotReady("transformation failed: " + err.Error())
+		return nil, err
+	}
+
+	// create installer set
+	tis := makeInstallerSet(td, manifest, r.releaseVersion)
+	createdIs, err := r.operatorClientSet.OperatorV1alpha1().TektonInstallerSets().
+		Create(ctx, tis, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return createdIs, nil
+}
+
+func makeInstallerSet(td *v1alpha1.TektonDashboard, manifest mf.Manifest, releaseVersion string) *v1alpha1.TektonInstallerSet {
+	ownerRef := *metav1.NewControllerRef(td, td.GetGroupVersionKind())
+	return &v1alpha1.TektonInstallerSet{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: fmt.Sprintf("%s-", v1alpha1.DashboardResourceName),
+			Labels: map[string]string{
+				createdByKey: createdByValue,
+			},
+			Annotations: map[string]string{
+				releaseVersionKey:  releaseVersion,
+				targetNamespaceKey: td.Spec.TargetNamespace,
+			},
+			OwnerReferences: []metav1.OwnerReference{ownerRef},
+		},
+		Spec: v1alpha1.TektonInstallerSetSpec{
+			Manifests: manifest.Resources(),
+		},
+	}
+}
+
+// ComputeHashOf generates an unique hash/string for the
+// object pass to it.
+func ComputeHashOf(obj interface{}) (string, error) {
```

Comment:
shall we move this into `pkg/reconciler/common` as this is a very generic function which is/could be used at multiple places.

Reply:
this is already moved to common pkg
we can use that in this file
common.ComputeHashOf

Reply:
Or it could go into a `hash` package, so that it becomes `hash.Compute(…)`. I know you all know, but `common` for a package name makes very little sense 😛

Reply:
Shall I move it to `hash` package then ?

Reply:
yes, 👍